### DISCUSSION TITLE
не работает активация

### DISCUSSION THREAD

**Discussion by miyagi-dojo** [quarckster/kodi.kino.pub#324] [General]
при попытке выполнить активацию выдается ошибка (см. лог)
пробовал ресетить авторизацию через настройки и активировать заново - не помогло
также пробовал установить kodi 18 и активировать через приложение версии 2.12.4 -  ошибка та же

просьба подсказать, как исправить если пробема с моей стороны 

**сведения о системе:**
raspberry pi 3
kodi build 20.3
LibreElec 11.0.6 (kernel: Linux 6.1.74)

лог (версия приложения 3.5.1 )
> 2024-01-30 11:39:38.537 T:753      info <general>: CActiveAESink::OpenSink - initialize sink
> 2024-01-30 11:39:38.537 T:753      info <general>: CAESinkALSA::Initialize - Attempting to open device "hdmi:CARD=vc4hdmi,DEV=0"
> 2024-01-30 11:39:38.544 T:753      info <general>: CAESinkALSA::Initialize - Opened device "hdmi:CARD=vc4hdmi,DEV=0,AES0=0x04,AES1=0x82,AES2=0x00,AES3=0x00"
> 2024-01-30 11:39:38.544 T:753      info <general>: CAESinkALSA::InitializeHW - Your hardware does not support AE_FMT_FLOAT, trying other formats
> 2024-01-30 11:39:38.544 T:753      info <general>: CAESinkALSA::InitializeHW - Using data format AE_FMT_S24NE3
> 2024-01-30 11:39:43.437 T:986     error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
>                                                     - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
>                                                    Error Type: <class 'urllib.error.URLError'>
>                                                    Error Contents: <urlopen error [Errno 104] Connection reset by peer>
>                                                    Traceback (most recent call last):
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1348, in do_open
>                                                      File "/usr/lib/python3.11/http/client.py", line 1282, in request
>                                                      File "/usr/lib/python3.11/http/client.py", line 1328, in _send_request
>                                                      File "/usr/lib/python3.11/http/client.py", line 1277, in endheaders
>                                                      File "/usr/lib/python3.11/http/client.py", line 1037, in _send_output
>                                                      File "/usr/lib/python3.11/http/client.py", line 975, in send
>                                                      File "/usr/lib/python3.11/http/client.py", line 1454, in connect
>                                                      File "/usr/lib/python3.11/ssl.py", line 517, in wrap_socket
>                                                      File "/usr/lib/python3.11/ssl.py", line 1075, in _create
>                                                      File "/usr/lib/python3.11/ssl.py", line 1346, in do_handshake
>                                                    ConnectionResetError: [Errno 104] Connection reset by peer
>                                                
>                                                    During handling of the above exception, another exception occurred:
>                                                    
>                                                    Traceback (most recent call last):
>                                                      File "/storage/.kodi/addons/video.kino.pub/addon.py", line 6, in <module>
>                                                        plugin.run()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/plugin.py", line 88, in run
>                                                        self.routing.dispatch(self.path)
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/routing.py", line 71, in dispatch
>                                                        view_func(**kwargs)
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/main.py", line 86, in login
>                                                        plugin.auth.get_token()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 218, in get_token
>                                                        self._activate()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 204, in _activate
>                                                        resp = self._get_device_code()
>                                                               ^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 127, in _get_device_code
>                                                        resp = self._make_request(payload)
>                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 85, in _make_request
>                                                        response = urllib.request.urlopen(
>                                                                   ^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 216, in urlopen
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 519, in open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 536, in _open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 496, in _call_chain
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1391, in https_open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1351, in do_open
>                                                    urllib.error.URLError: <urlopen error [Errno 104] Connection reset by peer>
>                                                    -->End of Python script error report<--

log версии 3.50

> 2024-01-30 12:25:17.991 T:1237    error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
>                                                     - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
>                                                    Error Type: <class 'urllib.error.URLError'>
>                                                    Error Contents: <urlopen error [Errno 104] Connection reset by peer>
>                                                    Traceback (most recent call last):
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1348, in do_open
>                                                      File "/usr/lib/python3.11/http/client.py", line 1282, in request
>                                                      File "/usr/lib/python3.11/http/client.py", line 1328, in _send_request
>                                                      File "/usr/lib/python3.11/http/client.py", line 1277, in endheaders
>                                                      File "/usr/lib/python3.11/http/client.py", line 1037, in _send_output
>                                                      File "/usr/lib/python3.11/http/client.py", line 975, in send
>                                                      File "/usr/lib/python3.11/http/client.py", line 1454, in connect
>                                                      File "/usr/lib/python3.11/ssl.py", line 517, in wrap_socket
>                                                      File "/usr/lib/python3.11/ssl.py", line 1075, in _create
>                                                      File "/usr/lib/python3.11/ssl.py", line 1346, in do_handshake
>                                                    ConnectionResetError: [Errno 104] Connection reset by peer
>                                                    
>                                                    During handling of the above exception, another exception occurred:
>                                                    
>                                                    Traceback (most recent call last):
>                                                      File "/storage/.kodi/addons/video.kino.pub/addon.py", line 6, in <module>
>                                                        plugin.run()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/plugin.py", line 88, in run
>                                                        self.routing.dispatch(self.path)
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/routing.py", line 71, in dispatch
>                                                        view_func(**kwargs)
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/main.py", line 86, in login
>                                                        plugin.auth.get_token()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 214, in get_token
>                                                        self._activate()
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 200, in _activate
>                                                        resp = self._get_device_code()
>                                                               ^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 123, in _get_device_code
>                                                        resp = self._make_request(payload)
>                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/storage/.kodi/addons/video.kino.pub/resources/lib/auth.py", line 81, in _make_request
>                                                        response = urllib.request.urlopen(
>                                                                   ^^^^^^^^^^^^^^^^^^^^^^^
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 216, in urlopen
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 519, in open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 536, in _open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 496, in _call_chain
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1391, in https_open
>                                                      File "/usr/lib/python3.11/urllib/request.py", line 1351, in do_open
>                                                    urllib.error.URLError: <urlopen error [Errno 104] Connection reset by peer>
>                                                    -->End of Python script error report<--

**Comment by stavultras** [comment]
Откройте в браузере https://api.srvkp.com/oauth2/device. Должен вернутся json с status: 405.
Либо откройте https://api.srvkp.com/v1, должно написать 404 Not Found.

Если не открывается, значит ваш провайдер блочит домен api.srvkp.com. Судя по логу - так и есть. Тут только VPN в raspberry без default gateway и прокидывать через VPN исключительно IP домена api.srvkp.com (172.66.43.49 и 172.66.40.207 на данный момент). Что бы остальной трафик не гонялся через VPN.

**Comment by b2vadim** [comment]
У меня тоже перестала работать активация на клиенте KODI. возможно сейчас проблема на стороне сервера активации.

DNS и IP

LibreELEC:

Host api.srvkp.com:443 was resolved.
IPv4: 8.47.69.6, 8.6.112.6
IPv6: (none)


PC в той же сети:

IPv4: 8.6.112.0, 8.47.69.0
IPv6: 2a06:98c1:3122:8000::, 2a06:98c1:3123:8000::


 Вывод: DNS резолвится корректно на обоих устройствах через разные IP.

На LibreELEC IPv6 нет, но соединение идёт через IPv4 — нормально.

IP совпадают с IP сервера, поэтому провайдер не блокирует доступ к самому адресу на уровне маршрутизации.

 TCP соединение

LibreELEC:

* Connected to api.srvkp.com (8.47.69.6) port 443


PC: соединение устанавливается и TLS handshake проходит.

 Вывод:

Провайдер не блокирует порт 443.

TCP соединение между LibreELEC и сервером проходит, поэтому проблема не в интернет-провайдере на уровне сети.

 TLS handshake

LibreELEC:

SSL_connect: SSL_ERROR_SYSCALL
no peer certificate available
unexpected eof while reading


PC:

TLSv1.3 handshake успешен
сертификат проверен


 На LibreELEC TLS рукопожатие обрывается сразу после Client Hello.

Это не связано с провайдером (пакеты доходят, TCP работает).

Это характерно для поведения сервера/Cloudflare: если клиент не поддерживает ожидаемые TLS extensions, ALPN, User-Agent, Cookie/JS challenge — сервер закрывает соединение.

На ПК handshake проходит → значит сервер доступен, провайдер не мешает.

 HTTP/Cloudflare 403 на ПК

Даже на ПК в одной сети сервер возвращает 403 HTML с challenge/блокировкой → это защита на стороне сервера (Cloudflare).

LibreELEC не выполняет JS и не хранит cookie → сервер закрывает TLS раньше → получаем EOF.

 OpenSSL версии

LibreELEC:

OpenSSL 3.5.4
TLSv1.3 поддерживается


TLS современный, но клиент (urllib / Python / Kodi) может не поддерживать ALPN, SNI, cookie/redirect — это не провайдерская проблема.

 Итог по данным

DNS работает, IP резолвятся → провайдер не блокирует.

TCP соединение устанавливается → провайдер не блокирует порт 443.

TLS handshake не проходит только на LibreELEC → проблема на стороне сервера/Cloudflare, конкретно на уровне фильтрации клиентского TLS.

На ПК в той же сети TLS и HTTP handshake проходят → подтверждение, что интернет-провайдер не виноват.

Вывод: проблема именно на стороне сервера (Cloudflare), а не у провайдера или в локальной сети.

Сервер «не доверяет» LibreELEC / urllib клиенту → обрывает TLS.

Любая попытка изменения DNS или IP провайдера здесь не поможет.

raspberry pi 2 , Libreelec 12.2 , video.kino.pub-3.6.0
